pkg/action/install.go

```diff
@@ -394,6 +394,26 @@ func (i *Install) RunWithContext(ctx context.Context, ch ci.Charter, vals map[st
 
 	// Bail out here if it is a dry run
 	if isDryRun(i.DryRunStrategy) {
+		// For server-side dry-run, validate resources against the API server
+		if i.DryRunStrategy == DryRunServer && len(resources) > 0 {
+			if len(toBeAdopted) == 0 {
+				_, err = i.cfg.KubeClient.Create(
+					resources,
+					kube.ClientCreateOptionServerSideApply(i.ServerSideApply, false),
+					kube.ClientCreateOptionDryRun(true),
+				)
+			} else {
+				_, err = i.cfg.KubeClient.Update(
+					toBeAdopted,
+					resources,
+					kube.ClientUpdateOptionServerSideApply(i.ServerSideApply, i.ForceConflicts),
+					kube.ClientUpdateOptionDryRun(true),
+				)
+			}
+			if err != nil {
+				return rel, err
+			}
```

Comment:
I think it is wanted/needed to split the adopted/created resources?

```suggestion
		if i.DryRunStrategy == DryRunServer {
			if len(toBeAdopted) > 0 {
				_, err := i.cfg.KubeClient.Update(
					toBeAdopted,
					resources,
					kube.ClientUpdateOptionServerSideApply(i.ServerSideApply, i.ForceConflicts),
					kube.ClientUpdateOptionDryRun(true),
				)
				if err != nil {
				    return rel, err
			    }
			}
			if len(resources) > 0 {
			    _, err := i.cfg.KubeClient.Create(
					resources,
					kube.ClientCreateOptionServerSideApply(i.ServerSideApply, false),
					kube.ClientCreateOptionDryRun(true),
				)
				if err != nil {
				    return rel, err
			    }
			}
```

I also suspect the error handling could be improved in the above perhaps. By not failing on the first error?

Reply:
@gjenkins8  thanks for finding the time for the review, I applied your feedback, could you please have a look when you have time? Thanks in advance.